Lecture2/Task5/Task5L2.java

```diff
@@ -0,0 +1,154 @@
+    }
+
+    private static void letterMessageValidation ( boolean checkRightMoveName , boolean checkRightMoveLastname , char[] nameArray , char[] lastnameArray , char[] starNameArray , char[] starLastnameArray ) {
+        if (checkRightMoveName && checkRightMoveLastname)
+            if (Arrays.equals ( nameArray , starNameArray ) || Arrays.equals ( lastnameArray , starLastnameArray ))
+                printWholeWord ( starNameArray , starLastnameArray );
+            else printRightMove ( starNameArray , starLastnameArray );
+        else if (checkRightMoveName)
+            if (Arrays.equals ( nameArray , starNameArray ))
+                printWholeWord ( starNameArray , starLastnameArray );
+            else printRightMove ( starNameArray , starLastnameArray );
+        else if (checkRightMoveLastname)
+            if (Arrays.equals ( lastnameArray , starLastnameArray ))
+                printWholeWord ( starNameArray , starLastnameArray );
+            else printRightMove ( starNameArray , starLastnameArray );
+        else printWrongMove ( );
+    }
+
+    private static void finishGameValidation ( boolean checkResultName , boolean checkResultLastname ) {
+        if (checkResultName && checkResultLastname)
+            System.out.print ( "\nWonderful!!! You did it!!! See you later...\n" );
+    }
+
+    private static boolean letterMoveValidation ( String choice , int arraylength , char[] array , char[] starArray , boolean checkRightMove ) {
+        for ( int i = 0 ; i < arraylength ; i++ )
+            if (choice.equalsIgnoreCase ( Character.toString ( array[ i ] ) )) {
+                starArray[ i ] = array[ i ];
+                checkRightMove = true;
+            }
+        return checkRightMove;
+    }
+
+    private static void letterProcessing ( char[] nameArray , char[] starNameArray , String choice , char[] lastnameArray , char[] starLastnameArray , List <String> listOfLetters ) {
+        boolean checkRightMoveName = false;
+        boolean checkRightMoveLastname = false;
+        if (!Arrays.equals ( nameArray , starNameArray ))
+            checkRightMoveName = letterMoveValidation ( choice , nameArray.length , nameArray , starNameArray , checkRightMoveName );
+        if (!Arrays.equals ( lastnameArray , starLastnameArray ))
+            checkRightMoveLastname = letterMoveValidation ( choice , lastnameArray.length , lastnameArray , starLastnameArray , checkRightMoveLastname );
+        listOfLetters.add ( choice );
+        letterMessageValidation ( checkRightMoveName , checkRightMoveLastname , nameArray , lastnameArray , starNameArray , starLastnameArray );
+    }
+
+    private static void wordMessageValidation ( boolean checkRightMove , char[] starNameArray , char[] starLastnameArray ) {
+        if (checkRightMove)
+            printWholeWord ( starNameArray , starLastnameArray );
+        else printWrongMove ( );
+    }
+
+    private static boolean wordMoveValidation ( char[] array , char[] starArray ) {
```

Comment:
no real sense in this method.  If this method did actual array copying, then yes - method is reasonable

Reply:
method name fixed